fabric-key-binding-api-v1/src/main/java/net/fabricmc/fabric/api/client/keybinding/v1/FabricKeyBinding.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.keybinding.v1;
+
+import java.util.Objects;
+import java.util.function.BooleanSupplier;
+
+import com.google.common.base.Preconditions;
+
+import net.minecraft.client.options.KeyBinding;
+import net.minecraft.client.util.InputUtil;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.impl.client.keybinding.StickyFabricKeyBinding;
+import net.fabricmc.fabric.mixin.client.keybinding.KeyCodeAccessor;
+
+/**
+ * Expanded version of {@link KeyBinding} for use by Fabric mods.
+ *
+ * <p>*ALL* instantiated FabricKeyBindings should be registered in
+ * {@link KeyBindingRegistry#register(FabricKeyBinding)}!
+ * </p>
+ *
+ * <pre><code>
+ * FabricKeyBinding.Builder builder = FabricKeyBinding.builder();
+ * FabricKeyBinding left = builder
+ * 			.id(new Identifier("example", "left"))
+ * 			.code(Keys.Left)
+ * 			.build();
+ * FabricKeyBinding right = builder
+ * 			.id(new Identifier("example", "right"))
+ * 			.code(Keys.Right)
+ * 			.build();
+ * KeyBindingRegistry.register(left);
+ * KeyBindingRegistry.register(right);
+ * </code></pre>
+ */
+public class FabricKeyBinding extends KeyBinding {
+	private final Identifier id;
+
+	protected FabricKeyBinding(Identifier id, String translationKey, InputUtil.Type type, int code, String category) {
+		super(translationKey, type, code, category);
+		this.id = id;
+	}
+
+	/**
+	 * Original identifier used to register this key.
+	 *
+	 * <p>May be different from the {@link #getId()}.</p>
+	 */
+	public Identifier getIdentifier() {
+		return id;
+	}
+
+	/**
+	 * Creates a new builder for constructing custom key bindings.
+	 */
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	/**
+	 * Returns the configured KeyCode assigned to the KeyBinding from the player's settings.
+	 *
+	 * @return configured KeyCode
+	 */
+	public InputUtil.KeyCode getBoundKey() {
```

Comment:
There are some naming inconsistencies between bound vs assigned.

Reply:
We have been using bound in the new yarn, where is assigned used

Reply:
Search the file for "assign", there are multiple uses both in code and docs